mono_repo/lib/src/commands/travis/travis_yaml.dart

```diff
@@ -209,72 +210,56 @@ List<Object> _calculateOrderedStages(RootConfig rootConfig) {
 
 /// Lists all the jobs, setting their stage, environment, and script.
 Iterable<Map<String, String>> _listJobs(
-  Iterable<TravisJob> jobs,
+  Iterable<CIJob> jobs,
   Map<String, String> commandsToKeys,
   Set<String> mergeStages,
 ) sync* {
-  final jobEntries = <_TravisJobEntry>[];
+  final jobEntries = <CIJobEntry>[];
 
   for (var job in jobs) {
     final commands =
         job.tasks.map((task) => commandsToKeys[task.command]).toList();
 
-    jobEntries.add(
-        _TravisJobEntry(job, commands, mergeStages.contains(job.stageName)));
+    jobEntries.add(CIJobEntry(job, commands));
   }
 
-  final groupedItems =
-      groupBy<_TravisJobEntry, _TravisJobEntry>(jobEntries, (e) => e);
+  // Group jobs by all of the values that would allow them to merge
```

Comment:
@jakemac53 – I was using that crazy equality mess to handle this case. Instead, I just inlined the logic.